File: Day5/validWordSquare.js
//422. Valid Word Square

var validWordSquare = function (words) {
    for (let i = 0; i < words.length; i++) {
        let res = ""
        for (let j = 0; j < words.length; j++) {
            res += words[j][i] || ''
        }

        console.log(res, words[i])

        if (res != words[i]) return false
    }

    return true
};

File: Day5/ValidWordSquare.java
    public ValidWordSquare{
       public boolean IsValidWordSquare(List<String> words) {
         for(int i=0;i<words.size();i++){
              char[] sc  = words.get(i).toCharArray();
              for(int j=0;j<sc.length;j++){
                  if(j >= words.size() || i >= words.get(j).length()) return false;
                  if(sc[j] != words.get(j).charAt(i)) return  false;
              }
          }        
          return true;
      }
    }
